local-cli/rnpm/core/src/config/android/index.js

```diff
@@ -33,11 +33,28 @@ exports.projectConfig = function projectConfigAndroid(folder, userConfig) {
   const packageFolder = userConfig.packageFolder ||
     packageName.replace(/\./g, path.sep);
 
-  const mainActivityPath = path.join(
+  // variable misnomer - Actually set to application path for 0.29+
+  let mainActivityPath = null;
+
+  mainActivityPath = path.join(
     sourceDir,
     userConfig.mainActivityPath || `src/main/java/${packageFolder}/MainActivity.java`
   );
 
+  // If MainApplication.java exists, use that instead
+  const mainApplicationPath = path.join(
+    sourceDir,
+    userConfig.mainApplicationPath || `src/main/java/${packageFolder}/MainApplication.java`
```

Comment:
I'd rename it from MainActivity/MainApplication to something more generic like `mainFilePath` or whatever is better. Don't like the idea of having `mainApplicationPath` along with `mainActivityPath`.